lib/galaxy/tools/cache.py

```diff
@@ -287,7 +289,8 @@ class ToolShedRepositoryCache:
     repos_by_tuple: Dict[Tuple[str, str, str], List[ToolConfRepository]]
 
     def __init__(self, session: install_model_scoped_session):
-        self.session = session()
+        engine = session.get_bind()
```

Comment:
Oh, this will crash the zergling on EU on 21.09. No traceback, unfortunately, the zergling just restarts.

Reply:
Hmm, ok, let's revert this on 21.09, 22.01 should drop uwsgi support anyway

Reply:
@bgruening This isn't part of 21.09, did you backport this ?

Reply:
Yes, I was trying if this is fixing https://github.com/galaxyproject/galaxy/issues/12988 and backported it but then the zerglings did not start.